FireRender.Maya.Src/MayaStandardNodesSupport/FileNodeConverter.cpp

```diff
@@ -13,27 +13,77 @@ limitations under the License.
 #include "FileNodeConverter.h"
 #include "FireMaya.h"
 
+#include <sstream>
+
 MayaStandardNodeConverters::FileNodeConverter::FileNodeConverter(const ConverterParams& params) : BaseConverter(params)
 {
 }
 
+void LoadAndAssignUdimImages(const MString& nodeName,  frw::Context context, frw::Image masterImage, const MString& udimPathPattern)
+{
+	MStringArray fileNames;
+
+	std::ostringstream ostream;
+	ostream << "source \"rprCmdRenderUtils.mel\"; geFileNodeUDIMFiles(\"" << nodeName.asChar() << "\")";
+
+	MStatus status = MGlobal::executeCommand(ostream.str().c_str(), fileNames);
+
+	size_t index = std::string(udimPathPattern.asChar()).find("<UDIM>");
+	size_t tagLength = std::string("UDIM").length();
+
+	if (index == std::string::npos)
+	{
+		MGlobal::displayWarning(MString("UDIM pattern is not recognized: ") + udimPathPattern);
+		return;
+	}
+
+	for (size_t i = 0; i < fileNames.length(); ++i)
+	{
+		std::string fileName = fileNames[(unsigned int) i].asChar();
+
+		rpr_uint tileIndex = std::stoi(fileName.substr(index, tagLength));
+		
+		frw::Image tile(context, fileName.c_str());
+		masterImage.SetUDIM(tileIndex, tile);
+	}
+}
+
+
 frw::Value MayaStandardNodeConverters::FileNodeConverter::Convert() const
 {
 	MPlug texturePlug = m_params.shaderNode.findPlug("computedFileTextureNamePattern");
 	MString texturePath = texturePlug.asString();
 
-	MPlug colorSpacePlug = m_params.shaderNode.findPlug("colorSpace");
+	MPlug uvTilingModePlug = m_params.shaderNode.findPlug("uvTilingMode");
+
+	const int fileNodeUdimMode = 3;
```

Comment:
please add comment what value 3 stand for

Reply:
Done